README.md

```diff
@@ -682,6 +682,66 @@ occurrence in `source` the function is applied to the current value of
 the behaviour and the value of the occurrence, the returned value
 becomes the next value of the behavior.
 
+#### `moment<A>(f: (sample: <B>(b: Behavior<B>) => B) => A): Behavior<A>``
+
+Constructs a behavior based on a function. At any point in time the value of
+the behavior is equal to the result of applying the function to a sampling
+function. The sampling function returns the current value of any behavior.
+
+`moment` is a powerful function that can do many things and sometimes it can do
+them in a way that is a lot easier than other functions. A typical usage of
+`moment` has the following form.
+
+```js
+moment((at) => {
+  ...
+})
+```
+
+Above, the `at` function above can be applied to any behavior and it will
+return the current value of the behavior. The following example adds together
+the values of three behaviors of numbers.
+
+```js
+const sum = moment((at) => at(aBeh) + at(bBeh) + at(cBeh));
+```
+
+The above could also be achieved with `lift`. However, `moment` can do many
+things that `lift` can't. For instance, `moment` can give better performance
```

Comment:
Why better performance?

Reply:
Thank you for the question.

If you write something like this:

```js
const lifted = lift((a, b, c, d) => a && b ? c : d, aB, bB, cB, dB);
```

Then the resulting behavior will _always_ depend on both `aB`, `bB`, `cB`, `dB`. This means that if any of them changes then the value of `lifted` will be recomputed. But, clearly we can see that if, for instance, `aB` is `false` then the function actually only uses `aB` and there is no need to recompute `lifted` if any of the other behaviors changes. However, Hareactive can't know that since the function is just a "black box".

If, on the other hand, we use `moment`:
```js
const momented = moment((at) => at(aB) && at(bB) ? at(cB) : at(dB));
```
Then Hareactive can simply check which behaviors are actually sampled inside the function passed to `moment`, and it uses this information to figure out which behaviors `momented` depends upon in any given time. This means that when `aB` is `false` then Hareactive can figure out that currently `momented` only depends on `atB` and there is no need to recompute it when any of the other behaviors changes.

Does that make sense or should I attempt to explain it better? Should I explain it better in the documentation?

Reply:
That explanation makes sense to me. I think you could put that right into the docs. (Also, the typo above!)